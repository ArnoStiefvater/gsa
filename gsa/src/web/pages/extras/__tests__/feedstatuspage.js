/* Copyright (C) 2020 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react';

import {rendererWith, wait, waitForElement} from 'web/utils/testing';

import FeedStatus from '../feedstatuspage';
import {Feed} from 'gmp/commands/feedstatus';

import Response from 'gmp/http/response';

const mockDate = new Date(1595833720475);

// set mockDate so the feed ages don't keep changing
global.Date.now = jest.fn(() => mockDate);

const nvtFeed = new Feed({
  name: 'Greenbone Community Feed',
  type: 'NVT',
  version: 202007241005,
});

const scapFeed = new Feed({
  name: 'Greenbone Community SCAP Feed',
  type: 'SCAP',
  version: 202007230130,
});

const certFeed = new Feed({
  name: 'Greenbone Community CERT Feed',
  type: 'CERT',
  version: 202007231003,
});

const gvmdDataFeed = new Feed({
  name: 'Greenbone Community GVMd Data Feed',
  type: 'GVMD_DATA',
  version: 202007221009,
});

const data = [nvtFeed, scapFeed, certFeed, gvmdDataFeed];

const xhr = {
  response: 'foo',
  responseText: 'bar',
  responseXML: 'ipsum',
};

const response = new Response(xhr, data);

const gmp = {
  feedstatus: {
    readFeedInformation: jest.fn(() => Promise.resolve(response)),
  },
  settings: {
    manualUrl: 'http://foo.bar',
  },
};

describe('Feed status page tests', () => {
  test('should render', async () => {
    const {render} = rendererWith({gmp, router: true});
    const {element, getAllByTestId} = render(<FeedStatus />);

    await wait();

    expect(element).toMatchSnapshot();

    // Should render all icons
    const icons = getAllByTestId('svg-icon');

    expect(icons.length).toBe(12);
    expect(icons[0]).toHaveTextContent('help.svg');
    expect(icons[0]).toHaveAttribute('title', 'Help: Feed Status');

    expect(icons[1]).toHaveTextContent('feed.svg');
    expect(icons[2]).toHaveTextContent('nvt.svg');
    expect(icons[3]).toHaveTextContent('cve.svg');
    expect(icons[4]).toHaveTextContent('cpe.svg');
    expect(icons[5]).toHaveTextContent('ovaldef.svg');
    expect(icons[6]).toHaveTextContent('cert_bund_adv.svg');
    expect(icons[7]).toHaveTextContent('dfn_cert_adv.svg');
    expect(icons[8]).toHaveTextContent('policy.svg');
    expect(icons[9]).toHaveTextContent('port_list.svg');
    expect(icons[10]).toHaveTextContent('report_format.svg');
    expect(icons[11]).toHaveTextContent('config.svg');
  });
});
