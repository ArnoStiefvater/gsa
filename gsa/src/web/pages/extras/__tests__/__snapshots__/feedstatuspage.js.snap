// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed status page tests should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c18 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -5px;
}

.c17 > * {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c17 > * {
  margin-left: 5px;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -5px;
}

.c19 > * {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c19 > * {
  margin-left: 5px;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -5px;
}

.c21 > * {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c21 > * {
  margin-left: 5px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c1 * {
  height: inherit;
  width: inherit;
}

.c6 {
  height: 50px;
  width: 50px;
  line-height: 50px;
}

.c6 * {
  height: inherit;
  width: inherit;
}

.c20 {
  height: 24px;
  width: 24px;
  line-height: 24px;
}

.c20 * {
  height: inherit;
  width: inherit;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin: 10px 0px;
  padding-bottom: 1px;
  border-bottom: 2px solid #c8d3d9;
  position: relative;
}

.c3 {
  margin: 0 0 1px 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-right: 5px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  word-break: break-all;
  min-width: 100px;
}

.c9 {
  border: 0;
  border-spacing: 0px;
  font-size: 12px;
  text-align: left;
  table-layout: auto;
  width: 100%;
}

.c9 th,
.c9 td {
  padding: 4px 10px;
  border-bottom: 1px solid #c8d3d9;
}

.c9 tfoot tr {
  background: #fff;
}

.c9 tfoot tr td {
  border-bottom: 1px solid #c8d3d9;
}

.c10 {
  background-color: #fff;
  color: #000;
  border-top: 1px solid #c8d3d9;
  font-weight: bold;
  width: 3rem;
}

.c12 {
  background-color: #fff;
  color: #000;
  border-top: 1px solid #c8d3d9;
  font-weight: bold;
  width: 21rem;
}

.c13 {
  background-color: #fff;
  color: #000;
  border-top: 1px solid #c8d3d9;
  font-weight: bold;
  width: 9rem;
}

.c14 {
  background-color: #fff;
  color: #000;
  border-top: 1px solid #c8d3d9;
  font-weight: bold;
  width: 7rem;
}

.c15 {
  background-color: #fff;
  color: #000;
  border-top: 1px solid #c8d3d9;
  font-weight: bold;
}

@media print {
  .c9 {
    border-collapse: collapse;
  }
}

@media screen {
  .c9 > tbody:nth-of-type(even),
  .c9 > tbody:only-of-type > tr:nth-of-type(even) {
    background: #eeeeee;
  }

  .c9 > tbody:not(:only-of-type):hover,
  .c9 > tbody:only-of-type > tr:hover {
    background: #c8d3d9;
  }
}

@media print {
  .c10 {
    color: #000;
    font-size: 1.2em;
    background-color: none;
    font-weight: bold;
  }
}

@media print {
  .c12 {
    color: #000;
    font-size: 1.2em;
    background-color: none;
    font-weight: bold;
  }
}

@media print {
  .c13 {
    color: #000;
    font-size: 1.2em;
    background-color: none;
    font-weight: bold;
  }
}

@media print {
  .c14 {
    color: #000;
    font-size: 1.2em;
    background-color: none;
    font-weight: bold;
  }
}

@media print {
  .c15 {
    color: #000;
    font-size: 1.2em;
    background-color: none;
    font-weight: bold;
  }
}

<div
  class="c0"
>
  <a
    href="http://foo.bar/en/web-interface.html#displaying-the-feed-status"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      class="c1"
      data-testid="svg-icon"
      title="Help: Feed Status"
    >
      <svg
        title="Help: Feed Status"
      >
        help.svg
      </svg>
    </span>
  </a>
  <section>
    <div
      class="c2 section-header"
    >
      <h2
        class="c3 c4"
      >
        <div
          class="c5"
        >
          <span
            class="c6"
            data-testid="svg-icon"
          >
            <svg>
              feed.svg
            </svg>
          </span>
        </div>
        <div
          class="c7"
        >
          Feed Status
        </div>
      </h2>
      <div
        class="c8"
      />
    </div>
  </section>
  <table
    class="c9"
  >
    <tbody
      class=""
    >
      <tr>
        <th
          class="c10"
        >
          <div
            class="c11"
          >
            Type
          </div>
        </th>
        <th
          class="c12"
        >
          <div
            class="c11"
          >
            Content
          </div>
        </th>
        <th
          class="c13"
        >
          <div
            class="c11"
          >
            Origin
          </div>
        </th>
        <th
          class="c14"
        >
          <div
            class="c11"
          >
            Version
          </div>
        </th>
        <th
          class="c15"
        >
          <div
            class="c11"
          >
            Status
          </div>
        </th>
      </tr>
      <tr>
        <td>
          <div
            class="c0"
          >
            NVT
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            <div
              class="c16"
            >
              <div
                class="c17"
                margin="5px"
              >
                <a
                  class="c18"
                  href="/nvts"
                >
                  <div
                    class="c16"
                  >
                    <div
                      class="c19"
                      margin="5px"
                    >
                      <span
                        class="c20"
                        data-testid="svg-icon"
                      >
                        <svg>
                          nvt.svg
                        </svg>
                      </span>
                      <span>
                        NVTs
                      </span>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            Greenbone Community Feed
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            20200724T1005
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            <div
              class="c16"
            >
              <div
                class="c21"
                margin="5px"
              >
                <strong>
                  2 days old
                </strong>
                <span />
              </div>
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            class="c0"
          >
            SCAP
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            <div
              class="c16"
            >
              <div
                class="c17"
                margin="5px"
              >
                <a
                  class="c18"
                  href="/cves"
                >
                  <div
                    class="c16"
                  >
                    <div
                      class="c19"
                      margin="5px"
                    >
                      <span
                        class="c20"
                        data-testid="svg-icon"
                      >
                        <svg>
                          cve.svg
                        </svg>
                      </span>
                      <span>
                        CVEs
                      </span>
                    </div>
                  </div>
                </a>
                <a
                  class="c18"
                  href="/cpes"
                >
                  <div
                    class="c16"
                  >
                    <div
                      class="c19"
                      margin="5px"
                    >
                      <span
                        class="c20"
                        data-testid="svg-icon"
                      >
                        <svg>
                          cpe.svg
                        </svg>
                      </span>
                      <span>
                        CPEs
                      </span>
                    </div>
                  </div>
                </a>
                <a
                  class="c18"
                  href="/ovaldefs"
                >
                  <div
                    class="c16"
                  >
                    <div
                      class="c19"
                      margin="5px"
                    >
                      <span
                        class="c20"
                        data-testid="svg-icon"
                      >
                        <svg>
                          ovaldef.svg
                        </svg>
                      </span>
                      <span>
                        OVAL Definitions
                      </span>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            Greenbone Community SCAP Feed
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            20200723T0130
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            <div
              class="c16"
            >
              <div
                class="c21"
                margin="5px"
              >
                <strong>
                  4 days old
                </strong>
                <span />
              </div>
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            class="c0"
          >
            CERT
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            <div
              class="c16"
            >
              <div
                class="c17"
                margin="5px"
              >
                <a
                  class="c18"
                  href="/certbunds"
                >
                  <div
                    class="c16"
                  >
                    <div
                      class="c19"
                      margin="5px"
                    >
                      <span
                        class="c20"
                        data-testid="svg-icon"
                      >
                        <svg>
                          cert_bund_adv.svg
                        </svg>
                      </span>
                      <span>
                        CERT-Bund Advisories
                      </span>
                    </div>
                  </div>
                </a>
                <a
                  class="c18"
                  href="/dfncerts"
                >
                  <div
                    class="c16"
                  >
                    <div
                      class="c19"
                      margin="5px"
                    >
                      <span
                        class="c20"
                        data-testid="svg-icon"
                      >
                        <svg>
                          dfn_cert_adv.svg
                        </svg>
                      </span>
                      <span>
                        DFN-CERT Advisories
                      </span>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            Greenbone Community CERT Feed
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            20200723T1003
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            <div
              class="c16"
            >
              <div
                class="c21"
                margin="5px"
              >
                <strong>
                  3 days old
                </strong>
                <span />
              </div>
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            class="c0"
          >
            GVMD_DATA
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            <div
              class="c16"
            >
              <div
                class="c17"
                margin="5px"
              >
                <a
                  class="c18"
                  href="/policies?filter=uuid%3Dc4b7c0cb-6502-4809-b034-8e635311b3e6%20"
                >
                  <div
                    class="c16"
                  >
                    <div
                      class="c19"
                      margin="5px"
                    >
                      <span
                        class="c20"
                        data-testid="svg-icon"
                      >
                        <svg>
                          policy.svg
                        </svg>
                      </span>
                      <span>
                        Compliance Policies
                      </span>
                    </div>
                  </div>
                </a>
                <a
                  class="c18"
                  href="/portlists?filter=uuid%3D33d0cd82-57c6-11e1-8ed1-406186ea4fc5%20uuid%3D4a4717fe-57d2-11e1-9a26-406186ea4fc5%20uuid%3D730ef368-57e2-11e1-a90f-406186ea4fc5%20"
                >
                  <div
                    class="c16"
                  >
                    <div
                      class="c19"
                      margin="5px"
                    >
                      <span
                        class="c20"
                        data-testid="svg-icon"
                      >
                        <svg>
                          port_list.svg
                        </svg>
                      </span>
                      <span>
                        Port Lists
                      </span>
                    </div>
                  </div>
                </a>
                <a
                  class="c18"
                  href="/reportformats?filter=uuid%3D5057e5cc-b825-11e4-9d0e-28d24461215b%20uuid%3Dc1645568-627a-11e3-a660-406186ea4fc5%20uuid%3D77bd6c4a-1f62-11e1-abf0-406186ea4fc5%20uuid%3Dc402cc3e-b531-11e1-9163-406186ea4fc5%20uuid%3Da3810a62-1f62-11e1-9219-406186ea4fc5%20uuid%3Da994b278-1f62-11e1-96ac-406186ea4fc5%20"
                >
                  <div
                    class="c16"
                  >
                    <div
                      class="c19"
                      margin="5px"
                    >
                      <span
                        class="c20"
                        data-testid="svg-icon"
                      >
                        <svg>
                          report_format.svg
                        </svg>
                      </span>
                      <span>
                        Report Formats
                      </span>
                    </div>
                  </div>
                </a>
                <a
                  class="c18"
                  href="/scanconfigs?filter=uuid%3Dd21f6c81-2b88-4ac1-b7b4-a2a9f2ad4663%20uuid%3D8715c877-47a0-438d-98a3-27c7a6ab2196%20uuid%3D085569ce-73ed-11df-83c3-002264764cea%20uuid%3Ddaba56c8-73ec-11df-a475-002264764cea%20uuid%3D2d3f051c-55ba-11e3-bf43-406186ea4fc5%20uuid%3Dbbca7412-a950-11e3-9109-406186ea4fc5%20"
                >
                  <div
                    class="c16"
                  >
                    <div
                      class="c19"
                      margin="5px"
                    >
                      <span
                        class="c20"
                        data-testid="svg-icon"
                      >
                        <svg>
                          config.svg
                        </svg>
                      </span>
                      <span>
                        Scan Configs
                      </span>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            Greenbone Community GVMd Data Feed
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            20200722T1009
          </div>
        </td>
        <td>
          <div
            class="c0"
          >
            <div
              class="c16"
            >
              <div
                class="c21"
                margin="5px"
              >
                <strong>
                  4 days old
                </strong>
                <span />
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
